lib/destination/types/table_config.go

```diff
@@ -56,30 +56,30 @@ func (d *DwhTableConfig) Columns() *columns.Columns {
 	return d.columns
 }
 
-func (d *DwhTableConfig) MutateInMemoryColumns(ctx context.Context, createTable bool, columnOp constants.ColumnOperation, cols ...columns.Column) {
+func (d *DwhTableConfig) MutateInMemoryColumns(createTable bool, columnOp constants.ColumnOperation, cols ...columns.Column) {
 	d.Lock()
 	defer d.Unlock()
 	switch columnOp {
 	case constants.Add:
 		for _, col := range cols {
 			d.columns.AddColumn(col)
 			// Delete from the permissions table, if exists.
-			delete(d.columnsToDelete, col.Name(ctx, nil))
+			delete(d.columnsToDelete, col.RawName())
```

Comment:
After a casual check it seems like `d.columnsToDelete` is only used for tests, and is `nil` otherwise. Do we need it?

Reply:
It's being used by internal functions to provide a read only version for external workloads e.g. `ReadOnlyColumnsToDelete`

Reply:
Ohh `AddColumnsToDelete` is mutating it, I had missed that before.